pkg/customresourcestate/config.go

```diff
@@ -206,3 +215,14 @@ func FromConfig(decoder ConfigDecoder, discovererInstance *discovery.CRDiscovere
 	}
 	return fn, nil
 }
+
+// configOverrides applies overrides to the configuration.
+func configOverrides(config *Metrics) {
+	for i := range config.Spec.Resources {
+		for j := range config.Spec.Resources[i].Metrics {
+
+			// Override the metric type to lowercase, so the internals have a single source of truth for metric type definitions.
```

Comment:
Could you perhaps add a comment on why this is necessary, just so we remember it in the future?